pandas/core/arrays/arrow/array.py

```diff
@@ -316,7 +318,7 @@ def _from_sequence(
 
     @classmethod
     def _from_sequence_of_strings(
-        cls, strings, *, dtype: Dtype | None = None, copy: bool = False
+        cls, strings, *, dtype: ExtensionDtype, copy: bool = False
```

Comment:
At least for arrow, `dtype=None` was supposed to mean "let `pa.array` infer the type, and now that can't be done?